hpx build failed on MacOS

When trying to build hpx on MacOS with 
```
cmake   -DCMAKE_CXX_COMPILER=clang++	\
	-DCMAKE_C_COMPILER=clang	\
	-DCMAKE_BUILD_TYPE=Debug	\
	-DHPX_WITH_MALLOC=system	\
	-DHPX_WITH_THREAD_LOCAL_STORAGE=ON ..
```
Result:
```
[  9%] Linking CXX shared library ../lib/libhpxd.dylib
Undefined symbols for architecture x86_64:
  "hpx_start::include_libhpx_wrap", referenced from:
      hpx::detail::run_or_start(hpx::util::function<int (boost::program_options::variables_map&), false> const&, boost::program_options::options_description const&, int, char**, std::__1::vector<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::allocator<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > >&&, hpx::util::unique_function<void (), false>, hpx::util::unique_function<void (), false>, hpx::runtime_mode, bool) in hpx_init.cpp.o
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
... Please describe your environment

  - HPX Version:1.2.0
  - Platform (compiler, OS): clang, MacOS

--- a/src/hpx_init.cpp
+++ b/src/hpx_init.cpp
@@ -290,7 +290,7 @@ namespace hpx_start
 {
     // Importing weak symbol from libhpx_wrap.a which may be shadowed by one present in
     // hpx_main.hpp.
-    HPX_SYMBOL_EXPORT __attribute__((weak)) extern bool include_libhpx_wrap;
+    HPX_SYMBOL_EXPORT __attribute__((weak)) bool include_libhpx_wrap = false;
 }
 
 #endif
@@ -611,7 +611,7 @@ namespace hpx
                             "using hpx::init. Exiting...\n";
                         return -1;
                     }
-                    #endif
+                #endif
 
                     std::cerr << "hpx::init: can't initialize runtime system "
                         "more than once! Exiting...\n";

